src/App.tsx

```diff
@@ -1,156 +1,429 @@
+/* eslint-disable max-len */
+/* eslint-disable no-console */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect, useState, FormEvent, useMemo,
+} from 'react';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { client } from './utils/client';
 
-const USER_ID = 0;
+const USER_ID = 10377;
+
+function getRandomNumber(): number {
+  const generatedNumbers: number[] = [];
+  const maxAttempts = 1001;
+
+  for (let i = 0; i < maxAttempts; i += 1) {
+    const randomNumber = Math.floor(Math.random() * 1001);
+
+    if (!generatedNumbers.includes(randomNumber)) {
+      generatedNumbers.push(randomNumber);
+
+      return randomNumber;
+    }
+  }
+
+  return 0;
+}
 
 export const App: React.FC = () => {
+  const [todo, setTodo] = useState<Todo[]>([]);
+  const [selectedTab, setSelectedTab] = useState('All');
+  const [inputValue, setInputValue] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [isThereActiveTodo, setIsThereActiveTodo] = useState(false);
+  const [isThereCompletedTodos, setIsThereCompletedTodos] = useState(false);
+  const [isAddingTodoAllowed, setIsAddingTodoAllowed] = useState(false);
+  const [isDeleteingTodoAllowed, setIsDeleteingTodoAllowed] = useState(false);
```

Comment:
You don't need to implement adding, deleting, and updating todos in this part of the task. It doesn't work correctly for now anyway. So you can remove it. Follow the requirements, please 
<img width="750" alt="image" src="https://github.com/mate-academy/react_todo-app-loading-todos/assets/99403198/68f76f1a-903f-43b4-9bed-8e6e929a1c7d">

Reply:
Not fixed. In this task only loading, filtering and errors should be implemented

Reply:
Hmm, still not fixed. Do you have any blockers here? Feel free to ask questions in the chat. Please, implement only the required tasks. You'll have the possibility to implement adding, deleting, and editing in the next parts of this task. There will be checklists and you'll be able to check yourself and find some hints. 
For now, it doesn’t work correctly. So this is the reason for rejection.
<img width="825" alt="image" src="https://github.com/mate-academy/react_todo-app-loading-todos/assets/99403198/13686c4c-cbd2-4403-a42e-45e4bcaaa430">